### DISCUSSION TITLE
Post IP address with custom fields via REST API

### DISCUSSION THREAD

**Discussion by astiks-io** [netbox-community/netbox#17254] [Help Wanted!]
I want to POST an IP with some custom fields.
SwaggerUI generated following request:
```
curl -X 'POST' \
  'https://172.21.0.242/api/ipam/ip-addresses/' \
  -H 'accept: application/json' \
  -H 'Content-Type: multipart/form-data' \
  -H 'X-CSRFTOKEN: lV9H5gy3HNkF2XakAfolGs3F5NX39L6NcJiANgr43FyKaA2w5f3two1vdQm8FnzV' \
  -F 'custom_fields={
  "test1": "123",
  "test2": "456"
}' \
  -F 'address=1.1.1.1/24'
```
But in response received `400 Error: Bad Request`:
```
{
  "custom_fields": [
    "Invalid data format. Custom field data must be passed as a dictionary mapping field names to their values."
  ]
}
```

Custom_fields value was filled this way:

![изображение](https://github.com/user-attachments/assets/f10b8d58-0bfc-4065-b9db-4a387b6d9a58)

Where did I go wrong?

**Comment by chris240189** [comment]
what are the types and names of the custom fields you are trying to write to?
